### PR TITLE
Fix flakiness of qos/test_qos_dscp_mapping.py

### PR DESCRIPTION
**Pull request by vkjammala-arista**
### Description of PR

Summary:
Fix flakiness of qos/test_qos_dscp_mapping.py::TestQoSSaiDSCPQueueMapping_IPIP_Base::test_dscp_to_queue_mapping_uniform_mode

Fixes # [aristanetworks/sonic-qual.msft#172](https://github.com/aristanetworks/sonic-qual.msft/issues/172)

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [x] 202311
- [x] 202405

#### What is the motivation for this PR?
In some of the cases, after sending packets (2000) the queue counter value is not reflecting correct value (counter value is less than expected) and thus lead to test failure. In the issue state, reading the counter value (in breakpoint) again shows the correct value. 

Sample output in failure case:
```
 3238         "  Inner Packet DSCP Value    Expected Egress Queue    Egress Queue Count  Result                              Actual Egress Queue\n",
 3239         "-------------------------  -----------------------  --------------------  --------------------------------  ---------------------\n",
 3240         "                        0                        1                   865  FAILURE - DUT POLL FAILURE                           -1\n",
 3241         "                        1                        1                  2000  SUCCESS                                               1\n",
 3242         "                        2                        1                  1022  FAILURE - DUT POLL FAILURE                           -1\n",
 3243         "                        3                        3                  2000  SUCCESS                                               3\n",
 3244         "                        4                        4                  1396  FAILURE - DUT POLL FAILURE                           -1\n",
 3245         "                        5                        1                  2000  SUCCESS                                               1\n",
 3246         "                        6                        1                  1449  FAILURE - DUT POLL FAILURE                           -1\n",
 3247         "                        7                        1                  2000  SUCCESS                                               1\n",
 3248         "                        8                        0                  1909  FAILURE - INCORRECT PACKET COUNT                      0\n",
 3249         "                        9                        1                  2000  SUCCESS                                               1\n",
...
```

#### How did you do it?
1) Updated the test to wait for atleast 10s (which is hardware counter polling time) before reading the queue counters.
```
$ counterpoll show | grep -i queue_stat
QUEUE_STAT                  default (10000)     enable
```
2) Updated the logic to re-poll the counters if the egress packet count is not as expected.

#### How did you verify/test it?
Stressed the test with fix on `Arista-7260CX3-D108C8`. Test is passing consistently with the fix.

### CODE REVIEW

**Comment by wsycqyz** [comment]
Do we need this PR in 2024 or 202311 branch?

**Comment by wsycqyz** [comment]
@XuChen-MSFT  FYI

**Comment by vkjammala-arista** [comment]
> Do we need this PR in 2024 or 202311 branch?

Yes, we need this PR in 202405 and 202311 branches.

**Review by wsycqyz — APPROVED** [review]

**Comment by wsycqyz** [comment]
I am ok with this change.

File: tests/qos/test_qos_dscp_mapping.py [hunk 1]
```diff
@@ -280,9 +278,12 @@ def _run_test(self,
             if packet_egressed_success:
                 dut_egress_port = get_dut_pair_port_from_ptf_port(duthost, tbinfo, dst_ptf_port_id)
                 pytest_assert(dut_egress_port, "No egress port on DUT found for ptf port {}".format(dst_ptf_port_id))
+                # Wait for the queue counters polling.
```
Comment (StormLiangMS) [inline tests/qos/test_qos_dscp_mapping.py:281]: @vivekverma-arista should use wait until here?

Comment (vkjammala-arista) [reply tests/qos/test_qos_dscp_mapping.py]: Thanks for the suggestion, updated the code to use `wait_until`.

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**